mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/AttachmentLoader.java

```diff
@@ -46,21 +47,24 @@ public AttachmentLoader(CassandraAttachmentMapper attachmentMapper) {
         this.attachmentMapper = attachmentMapper;
     }
 
-    public CompletableFuture<Stream<SimpleMailboxMessage>> toMailboxMessageWithAttachments(
-                CompletableFuture<Stream<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>>> messageRepresentations) {
+    public CompletableFuture<Stream<SimpleMailboxMessage>> addAttachmentToMessages(Stream<Pair<MessageWithoutAttachment,
+            Stream<MessageAttachmentRepresentation>>> messageRepresentations, MessageMapper.FetchType fetchType) {
 
-        return FluentFutureStream.of(messageRepresentations)
-            .thenComposeOnAll(pair -> getAttachments(pair.getRight().collect(Guavate.toImmutableList()))
-                    .thenApply(attachments -> Pair.of(pair.getLeft(), attachments)))
-            .map(pair ->
-                pair.getLeft()
-                    .toMailboxMessage(pair.getRight()
-                        .stream()
-                        .collect(Guavate.toImmutableList())))
-            .completableFuture();
+        if (fetchType == MessageMapper.FetchType.Body || fetchType == MessageMapper.FetchType.Full) {
+            return FluentFutureStream.of(
+                messageRepresentations
+                    .map(pair -> getAttachments(pair.getRight().collect(Guavate.toImmutableList()))
+                        .thenApply(attachments -> pair.getLeft().toMailboxMessage(attachments))))
+                .completableFuture();
+        } else {
+            return CompletableFuture.completedFuture(messageRepresentations
+                .map(pair -> pair.getLeft()
+                    .toMailboxMessage(ImmutableList.of())));
```

Comment:
It's a bit out of the scope of this PR but how can you make the distinction between a SimpleMailboxMessage that have no attachment and a SimpleMailboxMessage for which we did not fetch the attachment! Both will have a empty list for attachment.
Shouldn't we use Optional for that.
I am not asking you to fix that in this PR if you agree with me but maybe we should at least create a ticket for that.